examples/erc20/src/lib.rs

```diff
@@ -0,0 +1,298 @@
+            return Ok(Transfer {
+                from,
+                to,
+                amount
+            });
+        }
+        Err(Error::InsufficientFunds { address: from })
+    }
+
+    /// allowance
+    pub fn approve(&mut self, ctx: &Context, spender: Address, amount: u64) -> Result<Approval> {
+        let allowances = match self.allowed.entry(ctx.sender()) {
+            Entry::Vacant(ve) => ve.insert(HashMap::new()),
+            Entry::Occupied(oe) => oe.into_mut(),
+        };
+        allowances.insert(spender, amount);
+
+        let approval = Approval {
+            sender: ctx.sender(),
+            spender,
+            amount,
+        };
+
+        Event::emit(&approval);
+
+        Ok(approval.clone())
+    }
+
+    /// read allowance
+    pub fn allowance(&mut self, ctx: &Context, spender: Address) -> Result<u64> {
+        if !self.allowed.contains_key(&ctx.sender()) {
+            return Ok(0u64);
+        }
+        let allowances = self.allowed.get_mut(&ctx.sender()).unwrap();
+        if !allowances.contains_key(&spender) {
+            return Ok(0u64);
+        }
+        Ok(*allowances.get(&spender).unwrap())
+    }
+
+    /// transfer from a given account up to the given allowance
+    pub fn transfer_from(
+        &mut self,
+        _ctx: &Context,
+        from: Address,
+        spender: Address,
+        amount: u64,
+    ) -> Result<Transfer> {
+        let allowances = self.allowed.get_mut(&from).unwrap();
+        // if the spender is not in the list of addresses that are approved for automatic
+        // withdrawal by the from address, then nothing can be done
+        if !allowances.contains_key(&spender) {
+            return Err(Error::NoAllowanceGiven { from, to: spender });
+        }
+        let allowance = *allowances.get(&spender).unwrap();
+        // err if request is higher than allowance
+        if allowance < amount {
+            return Err(Error::RequestExceedsAllowance { amount, allowance });
+        }
+        if _transfer(&mut self.accounts, from, spender, amount) {
+            allowances.insert(spender, allowance - amount);
+            return Ok(Transfer {
+                from,
+                to: spender,
+                amount,
+            });
+        }
+        Err(Error::InsufficientFunds { address: from })
+    }
+}
+
+impl ERC20Token {
+    /// mint new tokens
+    pub fn mint(&mut self, ctx: &Context, amount: u64) -> Result<()> {
+        if !self.admins.contains(&ctx.sender()) {
+            return Err(Error::AdminPrivilegesRequired);
+        }
+        self.total_supply = self.total_supply + amount;
+        Ok(())
+    }
+
+    /// burn tokens from a given account
+    pub fn burn(&mut self, ctx: &Context, from: Address, amount: u64) -> Result<()> {
+        if !self.admins.contains(&ctx.sender()) {
+            return Err(Error::AdminPrivilegesRequired);
+        }
+        if self.admins.contains(&from) {
+            // no-op
+            return Ok(());
```

Comment:
why aren't admins allowed to burn?

Reply:
Resolved. Admins can burn now.. and crash before that :)